.github/workflows/ci.yml

```
@@ -68,6 +68,9 @@ jobs:
       - uses: r7kamura/rust-problem-matchers@v1.1.0
       - name: Run api tests
         run: cargo test -p bootloader_api
+      - name: Run bootloader common tests
+        if: runner.os == 'Linux'
```

The `x86_64` crate fails to compile under MacOs. I assume that is because macs use ARM architecture. Not quite sure, what the reason is.

---

Yup, `x86_64`'s default features include the `instructions` feature, which only compiles on `x86_64`.
```suggestion
        if: runner.arch == 'x64'
```